### PR TITLE
Apim 5455 v4 doc access controls

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-5455

## Description

Add creating a page with access controls + updating a page with access controls to MAPI-v2.

Next PR: 
- Implement FE for v4 documentation

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/RoleQueryServiceImpl.java [hunk 1]
```diff
@@ -68,4 +71,16 @@ public Optional<Role> findApplicationRole(String name, ReferenceContext referenc
             throw new TechnicalDomainException("An error occurs while trying to find application role", e);
         }
     }
+
+    @Override
+    public Set<Role> findByIds(Set<String> ids) {
+        if (Objects.isNull(ids)) {
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/membership/RoleQueryServiceImpl.java]: Also true if is empty

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/inmemory/RoleQueryServiceInMemory.java [hunk 2]
```diff
@@ -62,6 +65,15 @@ public Optional<Role> findApplicationRole(String name, ReferenceContext referenc
             .findFirst();
     }
 
+    @Override
+    public Set<Role> findByIds(Set<String> ids) {
+        if (Objects.isNull(ids)) {
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/inmemory/RoleQueryServiceInMemory.java]: Also true if is empty

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/model/Page.java [hunk 3]
```diff
@@ -54,15 +54,15 @@ public class Page {
     // Non-Folder attributes
     private String content;
     private boolean homepage;
+    private boolean excludedAccessControls;
+    private Set<AccessControl> accessControls;
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/model/Page.java]: Note: I think you could apply access control on folders too

**Review by phiz71 — CHANGES_REQUESTED** [review]
A few comments

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/model/Page.java (hunk 3 above)
Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/model/Page.java]: Ok I'll move it to the general attributes. The use case code applies to folders as well by default 👍

**Review by phiz71 — APPROVED** [review]

**Pull request merged**